pkg/controllers/nodeoverlay/validation/controller.go

```diff
@@ -0,0 +1,205 @@
+	// // Check if the requirements are compatible for each instance type
+	nodePoolList := &v1.NodePoolList{}
+	err := c.kubeClient.List(ctx, nodePoolList)
+	if err != nil {
+		return false, fmt.Errorf("listing nodepool, %w", err)
+	}
+	for _, np := range nodePoolList.Items {
+		its, err := c.cloudProvider.GetInstanceTypes(ctx, &np)
+		if err != nil {
+			return false, fmt.Errorf("listing instance types from nodepool, %w", err)
+		}
+		// To save on extra reconciliation, we validate all instance types for a NodePool
+		// as opposed to checking the instance that are defined with a NodePool requirement set
+		for _, it := range its {
+			if compatibleInstanceType(it, overlayRequirementsA, overlayRequirementsB) {
+				return true, nil
+			}
+		}
+	}
+
+	return false, nil
+}
+
+// compatibleInstanceType will validate if an instance type and its offerings offerings are compatible with two node overlay requirements.
+func compatibleInstanceType(it *cloudprovider.InstanceType, overlayReqA scheduling.Requirements, overlayReqB scheduling.Requirements) bool {
+	if it.Requirements.Compatible(overlayReqA) == nil && it.Requirements.Compatible(overlayReqB) == nil {
+		if overlayReqA.Keys().HasAny(v1.WellKnownLabelsForOfferings.UnsortedList()...) && overlayReqB.Keys().HasAny(v1.WellKnownLabelsForOfferings.UnsortedList()...) {
+			for _, of := range it.Offerings {
+				if of.Requirements.Compatible(overlayReqA) == nil && of.Requirements.Compatible(overlayReqB) == nil {
+					return true
+				}
+			}
+		} else {
+			return true
+		}
+	}
+	return false
+}
+
+// isConflictingOverlay validates if a the price, priceAdjustment, and capacity field all alter the same resources
+func (c *Controller) isConflictingOverlay(overlayOne v1alpha1.NodeOverlay, overlayTwo v1alpha1.NodeOverlay) bool {
+	conflictingResources := findConflictingResources(overlayOne.Spec.Capacity, overlayTwo.Spec.Capacity)
+	return lo.FromPtr(overlayOne.Spec.PriceAdjustment) != lo.FromPtr(overlayTwo.Spec.PriceAdjustment) ||
+		lo.FromPtr(overlayOne.Spec.Price) != lo.FromPtr(overlayTwo.Spec.Price) ||
+		len(conflictingResources) != 0
+}
+
+func (c *Controller) updateOverlayStatuses(ctx context.Context, overlayList []v1alpha1.NodeOverlay, overlaysWithConflict []string, overlayWithRuntimeValidationFailure map[string]error) error {
+	for _, overlay := range overlayList {
+		stored := overlay.DeepCopy()
+		overlay.StatusConditions().SetTrue(v1alpha1.ConditionTypeValidationSucceeded)
+		if err, ok := overlayWithRuntimeValidationFailure[overlay.Name]; ok {
+			overlay.StatusConditions().SetFalse(v1alpha1.ConditionTypeValidationSucceeded, "RuntimeValidation", err.Error())
+		} else if lo.Contains(overlaysWithConflict, overlay.Name) {
+			overlay.StatusConditions().SetFalse(v1alpha1.ConditionTypeValidationSucceeded, "Conflict", conflictMessage)
+		}
+
+		if !equality.Semantic.DeepEqual(stored, overlay) {
+			if err := c.kubeClient.Status().Patch(ctx, &overlay, client.MergeFrom(stored)); err != nil {
```

Comment:
Should an error in patching one overlap cause us to fail all of them?

Reply:
Also, I think we need to check the `client.IgnoreNotFound()` on L187 to avoid a NotFound error from causing us to exit early for all NodeOverlays